middleware.go

```diff
@@ -192,9 +192,13 @@ func createHTTPRequest(c *Client, r *Request) (err error) {
 			r.RawRequest, err = http.NewRequest(r.Method, r.URL, nil)
 		}
 	} else {
-		// fix data race: must deep copy.
-		bodyBuf := bytes.NewBuffer(append([]byte{}, r.bodyBuf.Bytes()...))
-		r.RawRequest, err = http.NewRequest(r.Method, r.URL, bodyBuf)
+		// deep copy
+		bodyCopy := acquireBuffer()
```

Comment:
I don't see where you release the acquired buffer

Reply:
I guess there is no way to release it, do you have any ideas?